### PR TITLE
[ERE-2151] Clinicians should be able to manage the users of all their visible patients if they have the required permissions

### PR DESCRIPTION
**Pull request by ppettitau**

### CODE REVIEW

**Comment by ppettitau** [comment]
Should I revert this previous commit which addresses an issue that would be superseded by this updated `get_by_user` code?
https://github.com/eresearchqut/trrf/commit/18055039e3a7282138c375eb5aaff35ba6ad8197

File: rdrf/registry/groups/models.py [hunk 1]
```diff
@@ -96,11 +95,26 @@ def get_by_user(self, staff_user):
...
 
         if staff_user.is_staff:
+
+            or_filters = [Q(working_groups__in=staff_user.working_groups.all()),
+                          Q(working_groups__isnull=True)]
+
+            if staff_user.is_clinician:
+                # Clinicians have special logic that potentially grants them access to users
+                # outside their working groups
+                from registry.patients.models import Patient
```
Comment (ScriptSmith) [inline rdrf/registry/groups/models.py:105]: local import

**Review by ScriptSmith — APPROVED** [review]

File: rdrf/registry/groups/models.py (hunk 1 above)
Comment (ppettitau) [reply rdrf/registry/groups/models.py:105]: It has to be local this time, otherwise it causes a circular dependency.

**Review by Lexiel46 — APPROVED** [review]

**Review by ScriptSmith — APPROVED** [review]
Reverting doesn't seem to matter, but up to you :+1:

**Pull request merged**